graphql/schema/types/providers.graphql

```diff
@@ -0,0 +1,236 @@
+    alertableInstanceTypes: [String!]!
+    allowedRegions: [String!]!
+    maxVolumeSizePerUser: Int!
+    pod: AWSPodConfigInput!
+    accountRoles: [AWSAccountRoleMappingInput]
+    IPAMPoolID: String!
+    elasticIPUsageRate: Float!
+}
+
+type AWSConfig {
+    ec2Keys: [EC2Key]
+	subnets: [Subnet]	
+	parserProject: ParserProjectS3Config
+	persistentDNS: PersistentDNSConfig
+	defaultSecurityGroup: String
+	allowedInstanceTypes: [String!]!
+	alertableInstanceTypes: [String!]!
+	allowedRegions: [String!]!
+	maxVolumeSizePerUser: Int!
+	pod: AWSPodConfig!
+	accountRoles: [AWSAccountRoleMapping]
+	IPAMPoolID: String!
+	elasticIPUsageRate: Float!
+}
+
+input AWSPodConfigInput {
+    role: String @redactSecrets
+    region: String
+    ecs: ECSConfigInput
+    secretsManager: SecretsManagerConfigInput
+}
+
+type AWSPodConfig {
+    role: String @requireAdmin
+    region: String
+    ecs: ECSConfig
+    secretsManager: SecretsManagerConfig
+}
+
+input AWSVPCConfigInput {
+    subnets: [String]
+    securityGroups: [String] @redactSecrets
+}
+
+type AWSVPCConfig {
+    subnets: [String]
+    securityGroups: [String] @requireAdmin
+}
+
+input CloudProvidersInput {
+    aws: AWSConfigInput
+    docker: DockerConfigInput
+}
+
+type CloudProviders {
+    aws: AWSConfig
+    docker: DockerConfig
+}
+
+input ContainerPoolInput {
+  id: String!
+  distro: String!
+  maxContainers: Int!
+  port: Int!
+}
+
+type ContainerPool {
+  id: String!
+  distro: String!
+  maxContainers: Int!
+  port: Int!
+}
+
+
+input ContainerPoolsConfigInput {
+    pools: [ContainerPoolInput]
+}
+
+type ContainerPoolsConfig {
+    pools: [ContainerPool]
+}
+
+input DockerConfigInput {
+    apiVersion: String
+}
+
+type DockerConfig {
+    apiVersion: String
+}
+
+input EC2KeyInput {
+    name: String!
+    region: String!
+    key: String! @redactSecrets
+    secret: String! @redactSecrets
+}
+
+type EC2Key {
+    name: String!
+    region: String!
+    key: String! @requireAdmin
+    secret: String! @requireAdmin
+}
+
+input ECSConfigInput {
+    maxCPU: Int
+    maxMemoryMb: Int
+    taskDefinitionPrefix: String
+    taskRole: String
+    executionRole: String
+    logRegion: String
+    logGroup: String
+    logStreamPrefix: String
+    AWSVPC: AWSVPCConfigInput
+    clusters: [ECSClusterConfigInput]
+    capacityProviders: [ECSCapacityProviderInput]
+    allowedImages: [String]
+}
+
+type ECSConfig {
+    maxCPU: Int
+    maxMemoryMb: Int
+    taskDefinitionPrefix: String
+    taskRole: String
+    executionRole: String
+    logRegion: String
+    logGroup: String
+    logStreamPrefix: String
+    AWSVPC: AWSVPCConfig
+    clusters: [ECSClusterConfig]
+    capacityProviders: [ECSCapacityProvider]
+    allowedImages: [String]
+}
+
+input ECSCapacityProviderInput {
+    name: String
+    os: String
+    arch: String
+    windowsVersion: String
+}
+
+type ECSCapacityProvider {
+    name: String
+    os: String
+    arch: String
+    windowsVersion: String
```

Comment:
can we define enums for `os`, `arch`, and `windowsVersion`? they should already exist in `globals.go`; I've copied them below

```
const (
	ECSOSLinux   = "linux"
	ECSOSWindows = "windows"
)

const (
	ECSArchAMD64 = "amd64"
	ECSArchARM64 = "arm64"
)

const (
	ECSWindowsServer2016 = "windows_server_2016"
	ECSWindowsServer2019 = "windows_server_2019"
	ECSWindowsServer2022 = "windows_server_2022"
)
```